server/xpub/entities/manuscript/resolvers.js

```diff
@@ -192,20 +192,19 @@ ${err}`,
           )
         }
       })
-
-      const saveFileStream = stream.pipe(progressReport)
-      await FileManager.putContent(
-        fileEntity,
-        saveFileStream,
-        { size: fileSize },
-      )
-
       // also send source file to conversion service
       const convertFileRequest = request.post(config.get('scienceBeam.url'), {
         qs: { filename },
         headers: { 'content-type': mimetype },
       })
+
+      // note: if a stream is piped to multiple destinations, ensure the
+      // pipes are set up in the same tick, or else chunks can be lost
+      const saveFileStream = stream.pipe(progressReport)
```

Comment:
Might be missing something - but how does this ensure it happens in the same tick?

Reply:
By "tick" I mean [event loop tick](https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/). The problem before was that we had an `await` between the two `pipe`s. Now we can guarantee that both `stream.pipe`s will be called before the stream starts flowing. A different stream is passed to `FileManager.putContents` so we don't need to worry about that one.